packages/server/src/sdk/app/rows/search.ts

```diff
@@ -115,3 +130,48 @@ export async function fetchView(
 ): Promise<Row[]> {
   return pickApi(tableId).fetchView(viewName, params)
 }
+
+async function getQueriableFields(
+  fields: string[],
+  table: Table
+): Promise<string[]> {
+  const handledTables = new Set<string>([table._id!])
+  const extractTableFields = async (
+    table: Table,
+    allowedFields: string[]
+  ): Promise<string[]> => {
+    const result = []
+    for (const field of Object.keys(table.schema).filter(f =>
+      allowedFields.includes(f)
+    )) {
+      const subSchema = table.schema[field]
+      if (subSchema.type === FieldType.LINK) {
+        if (handledTables.has(`${table._id}_${subSchema.tableId}`)) {
+          // avoid circular loops
+          continue
+        }
+        handledTables.add(`${subSchema.tableId}_${table._id}`)
```

Comment:
It's not obvious to me why the order is flipped here, `a_b` vs `b_a`. Is this intentional?

Reply:
It's to avoid circular loops. It's not so obvious, I am refactoring it and writing some tests for it